src/redisearch_rs/inverted_index/src/numeric.rs

```diff
@@ -560,12 +562,12 @@ impl Value {
         let u64_val = abs_val as u64;
 
         if u64_val as f64 == abs_val {
-            if u64_val <= 0b111 {
+            if value.is_sign_negative() {
+                Value::IntegerNegative(u64_val)
+            } else if u64_val <= 0b111 {
```

Comment:
Is this a bug fix or a clarity change?

Reply:
This is a bug fix. I also added a new test in https://github.com/RediSearch/RediSearch/pull/6851/commits/481b6792645ac7f566815f4b35909d4029b6792d